contracts/ERC7432/ERC7432.sol

```diff
@@ -113,8 +118,7 @@ contract ERC7432 is IERC7432 {
         address _grantor,
         address _grantee
     ) external view returns (bool) {
-        return  latestGrantees[_grantor][_tokenAddress][_tokenId][_role] == _grantee && roleAssignments[_grantor][_grantee][_tokenAddress][_tokenId][_role].expirationDate >
-            block.timestamp;
+        return latestGrantees[_grantor][_tokenAddress][_tokenId][_role] == _grantee && roleAssignments[_grantor][_grantee][_tokenAddress][_tokenId][_role].expirationDate > block.timestamp;
```

Comment:
`hasRole` and `hasUniqueRole` is only checking if the role is expired. We now need to check if the role was revoked prior to the expiration date as well

Reply:
This case is already covered by the original revokeRole logic, i added some tests to double checked it